src/main/java/org/spongepowered/common/data/processor/common/SpawnerUtils.java

```diff
@@ -0,0 +1,100 @@
+package org.spongepowered.common.data.processor.common;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityList;
+import net.minecraft.entity.passive.EntityPig;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.MobSpawnerBaseLogic;
+import net.minecraft.world.World;
+import org.spongepowered.api.entity.EntitySnapshot;
+import org.spongepowered.api.entity.EntityType;
+import org.spongepowered.api.entity.EntityTypes;
+import org.spongepowered.api.util.weighted.TableEntry;
+import org.spongepowered.api.util.weighted.WeightedObject;
+import org.spongepowered.api.util.weighted.WeightedSerializableObject;
+import org.spongepowered.api.util.weighted.WeightedTable;
+import org.spongepowered.common.entity.SpongeEntitySnapshot;
+import org.spongepowered.common.entity.SpongeEntitySnapshotBuilder;
+import org.spongepowered.common.registry.type.entity.EntityTypeRegistryModule;
+
+import java.util.Optional;
+
+public class SpawnerUtils {
+
+    public static Optional<Entity> createEntity(EntityType type, World world) {
```

Comment:
That method requires that a location be required - is Location(0,0,0) a reasonable default? This would also be carried across to the EntitySnapshot, so it will have a potentially unreasonable default spawn.